python/simulation.py

```diff
@@ -2276,7 +2310,9 @@ def using_real_fields(self):
         cond5 = not (cond3 or cond4 or self.k_point == Vector3())
         return not (self.force_complex_fields or cond1 or cond2 or cond5)
 
-    def initialize_field(self, cmpnt, amp_func):
+    def initialize_field(
+        self, cmpnt: int = None, amp_func: Callable[[Vector3Type], complex] = None
```

Comment:
Does the `complex` here allow the function to have real outputs?

Reply:
```suggestion
        self, cmpnt: int = None, amp_func: Callable[[Vector3Type], Union[float,complex]] = None
```

Reply:
Unfortunately, this change exceeds the 80-character line limit which is causing the pre-commit check involving black to fail. In the future, we need to be more careful about committing changes directly from GitHub.